### PR TITLE
WEBUI-879: select all remains active when unselecting individual docu…

### PR DESCRIPTION
**Pull request by rakeshkumar1019**
https://jira.nuxeo.com/browse/ELEMENTS-1527

### CODE REVIEW

File: ui/nuxeo-page-provider-display-behavior.js [hunk 1]
```diff
@@ -437,10 +437,15 @@ export const PageProviderDisplayBehavior = [
     },
 
     _toggleSelectAll() {
-      if (this.selectAllActive) {
+      if (this._isExcludedItemsNotEmpty && this.selectAllActive) {
         this.clearSelection();
+        this._isExcludedItemsNotEmpty = false;
+      } else if (this._isExcludedItemsNotEmpty === false && this.selectAllActive) {
```
Comment (swarnadipa-dev) [inline ui/nuxeo-page-provider-display-behavior.js]: this._isExcludedItemsNotEmpty === false can be written as !this._isExcludedItemsNotEmpty instead.

File: ui/nuxeo-data-table/iron-data-table.js [hunk 2]
```diff
@@ -231,7 +231,7 @@ import '../nuxeo-button-styles.js';
             <nuxeo-data-table-row header>
               <nuxeo-data-table-checkbox
                 style$="[[_computeSelectAllVisibility(selectionEnabled, selectAllEnabled, multiSelection)]]"
-                checked="[[selectAllActive]]"
+                checked="[[_isExcludedItemsNotEmpty]]"
```
Comment (Gabez0r) [inline ui/nuxeo-data-table/iron-data-table.js]: Perhaps we could do a simpler implementation, using a computed binding, with `selectAllActive` and `_excludedItems` dependencies? E.g.:
```suggestion
                checked="[[_isChecked(selectAllActive, _excludedItems)]]"
```
and then the `_isChecked` could do all the logics you do in `_onCheckBoxTap` and `_toggleSelectAll`?

Also, have you verified if the current solution fixes the issue for the other listings (e.g., data grid, data list)?

**Review by Gabez0r — COMMENTED** [review]
Looking good.

File: ui/nuxeo-data-table/iron-data-table.js (hunk 2 above)
Comment (rakeshkumar1019) [reply ui/nuxeo-data-table/iron-data-table.js]: @Gabez0r checked for data table.

Comment (rakeshkumar1019) [reply ui/nuxeo-data-table/iron-data-table.js]: let me do the test for data list.

**Review by poonamyadav252 — COMMENTED** [review]
Kindly try to pass UI checks.

**Pull request closed without merging**